packages/hooks/src/useCookieState/index.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useCallback } from 'react';
+import Cookies from 'js-cookie';
+
+function isFunction(obj: any): obj is Function {
+  return typeof obj === 'function';
+}
+
+export interface Options extends Cookies.CookieAttributes {
+  defaultValue?: CookieState | (() => CookieState);
+}
+
+export type CookieState = string | undefined;
+
+function useCookieState(
+  cookieKey: string,
+  options: Options = {},
+): [
+  CookieState,
+  (
+    newValue?: CookieState | ((prevState?: CookieState) => CookieState),
+    options?: Cookies.CookieAttributes,
+  ) => void,
+] {
+  const [state, setState] = useState<CookieState>(() => getDefaultValue());
+
+  // 获取defaultValue
+  function getDefaultValue(): CookieState {
+    const cookieValue = Cookies.get(cookieKey);
+    if (cookieValue !== undefined) return cookieValue;
+    if (isFunction(options.defaultValue)) return options.defaultValue();
+    return options.defaultValue;
+  }
```

Comment:
`getDefaultValue` 这个函数为啥不直接 inline 掉呢。。只有这一处使用
这样就可以直接用 ts 的类型推导了
```suggestion
  const [state, setState] = useState(() => {
  const cookieValue = Cookies.get(cookieKey);
    if (cookieValue !== undefined) return cookieValue;
    if (isFunction(options.defaultValue)) return options.defaultValue();
    return options.defaultValue;
  });
```

Reply:
👌